src/ngx_http_lua_consts.c

```diff
@@ -124,6 +124,9 @@ ngx_http_lua_inject_http_consts(lua_State *L)
     lua_setfield(L, -2, "HTTP_SEE_OTHER");
 #endif
 
+    lua_pushinteger(L, NGX_HTTP_PERMANENT_REDIRECT);
+    lua_setfield(L, -2, "HTTP_PERMANENT_REDIRECT");
```

Comment:
I think you should adjust the initial table size of `ngx` earlier to avoid table auto-growth (which is slow). Also, you need to update the tests covering the number of `ngx` table fields in the test suite (otherwise the test suite would be failing).